src/components/marker.js

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+import {Component, createElement} from 'react';
+import PropTypes from 'prop-types';
+import {PerspectiveMercatorViewport} from 'viewport-mercator-project';
+
+const propTypes = {
+  // Longitude of the anchor point
+  longitude: PropTypes.number.isRequired,
+  // Latitude of the anchor point
+  latitude: PropTypes.number.isRequired
+};
+
+const defaultProps = {
+};
+
+const contextTypes = {
+  viewport: PropTypes.instanceOf(PerspectiveMercatorViewport)
+};
+
+export default class Marker extends Component {
+
+  render() {
+    const {longitude, latitude} = this.props;
+
+    const [x, y] = this.context.viewport.project([longitude, latitude]);
+
+    return createElement('div', {
```

Comment:
Using Mapbox you create a htm node using `document.createElement`, wrap it in a Marker or Popup control class, then call `.addTo(map)` (https://www.mapbox.com/mapbox-gl-js/api/#popup#setdomcontent). The default Marker in Mapbox is just an empty div. Popup is essentially a Marker with additional features such as alignment, close button etc.